core/src/backend/render.rs

```diff
@@ -300,6 +300,11 @@ pub fn decode_jpeg(
     })
 }
 
+fn rgb5_component(compressed: u16, shift: u32) -> u8 {
+    let component = (compressed.overflowing_shr(shift).0) & 0x1F;
+    ((component * 255 + 15) / 31) as u8
+}
```

Comment:
Do we really need `overflowing_shr` since the shift values are known (0, 5, 10)? If we really care `wrapping_shr` is the same but doesn't return the tuple.

Reply:
Initially I was worried about garbage data potentially ruining a slightly valid image, but you're right - in this case we know the bounds and there's no way for this to be an issue.